src/features/session_replay/aggregate/index.js

```diff
@@ -0,0 +1,304 @@
+          session.state.sessionReplay = this.mode
+        }
+      }
+    }, this.featureName, this.ee)
+
+    // new handler for waiting for multiple flags.  will be useful if/when backend designs multiple flags, or for evaluating multiple feature flags simultaneously (stn vs sr)
+    this.waitForFlags(['sr']).then(([{ value }]) => {
+      this.initializeRecording(
+        value,
+        Math.random() < getConfigurationValue(this.agentIdentifier, 'session_replay.errorSampleRate'),
+        Math.random() < getConfigurationValue(this.agentIdentifier, 'session_replay.sampleRate')
+      )
+    })
+
+    drain(this.agentIdentifier, this.featureName)
+  }
+
+  /**
+   * Evaluate entitlements and sampling before starting feature mechanics, importing and configuring recording library, and setting storage state
+   * @param {boolean} entitlements - the true/false state of the "sr" flag from RUM response
+   * @param {boolean} errorSample - the true/false state of the error sampling decision
+   * @param {boolean} fullSample - the true/false state of the full sampling decision
+   * @returns {void}
+   */
+  async initializeRecording (entitlements, errorSample, fullSample) {
+    this.initialized = true
+    if (!entitlements) return
+
+    const { session } = getRuntime(this.agentIdentifier)
+    // if theres an existing session replay in progress, there's no need to sample, just check the entitlements response
+    // if not, these sample flags need to be checked
+    // if this isnt the FIRST load of a session AND
+    // we are not actively recording SR... DO NOT import or run the recording library
+    // session replay samples can only be decided on the first load of a session
+    // session replays can continue if already in progress
+    if (!session.isNew && session.state.sessionReplay === MODE.OFF) return
+
+    if (session.state.sessionReplay === MODE.FULL) this.mode = MODE.FULL
+    else if (session.isNew && fullSample) this.mode = MODE.FULL
+    else if (session.state.sessionReplay === MODE.ERROR) this.mode = MODE.ERROR
+    else if (errorSample) this.mode = MODE.ERROR
+    else return
+
+    // FULL mode records AND reports from the beginning, while ERROR mode only records (but does not report).
+    // ERROR mode will do this until an error is thrown, and then switch into FULL mode.
+    // If an error happened before we've gotten to this stage, we can simply just run in FULL mode.
+    if (this.mode === MODE.FULL) {
+      this.mode = MODE.FULL
+      // We only report (harvest) in FULL mode
+      this.scheduler.startTimer(this.harvestTimeSeconds)
+    }
+    // We record in FULL or ERROR mode
+
+    recorder = (await import(/* webpackChunkName: "recorder" */'rrweb')).record
+    this.startRecording()
+    const { gzipSync, strToU8 } = await import(/* webpackChunkName: "compressor" */'fflate')
+    gzipper = gzipSync
+    u8 = strToU8
+
+    this.isFirstChunk = !!session.isNew
+
+    session.state.sessionReplay = this.mode
```

Comment:
This should be moved earlier; see https://github.com/newrelic/newrelic-browser-agent/pull/532/files#r1199100432

Reply:
again, we just need to consider that we probably dont want to do this until we know rrweb made it to the page